docs/wallets.md

```diff
@@ -22,6 +22,19 @@ Supporting seed generation is a little more involved so the tradeoff between
 implementation complexity and user value is less clear, especially since the
 Codex provides users instructions on doing generation themselves.
 
+## Error Detection and Correction
+
+Wallets MUST support detection of errors using the codex32 checksum algorithm.
+Wallets MAY additionally support error correction; such wallets are referred to as "error-correcting wallets (ECWs)" and have additional requirements.
+
+An ECW
+
+* MUST support correction of up to 4 substitution errors;
```

Comment:
Yes, I'll write a reference implementation. I guess you are imagining that a wallet could support correcting up to 2 substitution via brute-force? Then what happens if it can't correct a string? It shouldn't say "uncorrectable" because the string may well be correctable, the wallet just has a poor implementation. So we would need extra text to discuss this case.

Better to just say "implement a standard ECC algorithm or don't bother".

Reply:
Yes, I did 2 substitutions by brute force.
It will say the checksum is invalid and won't display a candidate correction. Same as with full ECC. Don't claim an input is or is not "correctable", just display the closest edit distance valid candidate found, if any.

Even full ECC sometimes can't correct if the closest header found is invalid.